src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -114,5 +137,50 @@ private bool ShouldRestoreOnBuild
                 return packageRestoreConsent.IsAutomatic;
             }
         }
+
+        #region IVsUpdateSolutionEvents4
+
+        public void UpdateSolution_QueryDelayFirstUpdateAction(out int pfDelay)
+        {
+            // check if NuGet lock is already acquired by some other NuGet operation
+            if (LockService.IsLockHeld)
+            {
+                // delay build by setting pfDelay to non-zero
+                pfDelay = 1;
+            }
+            else
+            {
+                var isLockAcquired = LockService.TryAcquireLock(NuGetLockTimeout, out _nuGetLock);
```

Comment:
which is why I rather stick with `LockService.IsLockHeld` check instead of making `_nugetLock `null in dispose everytime and then checking for not null here... What you think?